### PR TITLE
Network controller - remove dead code

### PR DESCRIPTION
**Pull request by kumavis**
This simplifies the network controller by removing some dead code and refactoring a little

Fixes https://github.com/MetaMask/metamask-extension/issues/4147

### CODE REVIEW

File: app/scripts/controllers/network/network.js [hunk 1]
```diff
@@ -14,52 +14,41 @@ const {
...
     super()
 
-    this._networkEndpointVersion = OLD_UI_NETWORK_TYPE
-    this._networkEndpoints = getNetworkEndpoints(OLD_UI_NETWORK_TYPE)
-    this._defaultRpc = this._networkEndpoints[DEFAULT_NETWORK]
-
-    config.provider.rpcTarget = this.getRpcAddressForType(config.provider.type, config.provider)
+    // parse options
+    const providerConfig = opts.provider || defaultProviderConfig
+    console.log('providerStore:', providerConfig)
```
Comment (bitpshr) [inline app/scripts/controllers/network/network.js]: If this wasn't for debugging, should we use the `loglevel` module for consistency?

**Review by bitpshr — DISMISSED** [review]
Syntactically the changes make sense to me. Looks good, lots of red.

File: app/scripts/controllers/network/network.js (hunk 1 above)
Comment (kumavis) [reply app/scripts/controllers/network/network.js]: oops, yeah this was for debugging

**Review by frankiebee — APPROVED** [review]
lgtm would also like a review from @alextsg before merge

**Review by bitpshr — APPROVED** [review]

File: app/scripts/controllers/network/network.js [hunk 2]
```diff
@@ -14,52 +14,40 @@ const {
...
-  OLD_UI_NETWORK_TYPE,
-  DEFAULT_NETWORK,
+  LOCALHOST,
 } = require('./enums')
-const { getNetworkEndpoints } = require('./util')
+const LOCALHOST_RPC_URL = 'http://localhost:8545'
 const INFURA_PROVIDER_TYPES = [ROPSTEN, RINKEBY, KOVAN, MAINNET]
 
+const env = process.env.METAMASK_ENV
+const METAMASK_DEBUG = process.env.METAMASK_DEBUG
```
Comment (alextsg) [inline app/scripts/controllers/network/network.js]: nit: `const { METAMASK_ENV: env, METAMASK_DEBUG } = process.env`

**Review by alextsg — APPROVED** [review]

**Pull request merged**